src/app.py

```diff
@@ -1,62 +1,196 @@
+# app.py
 import os
 import pyromb
 from plot_catchment import plot_catchment
 import shapefile as sf
+from shapely.geometry import shape as shapely_shape
+import logging
+import json
+from typing import Any
+from pyromb.core.geometry.shapefile_validation import (
+    validate_shapefile_fields,
+    validate_shapefile_geometries,
+    validate_confluences_out_field,
+)
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
 
 DIR = os.path.dirname(__file__)
-REACH_PATH = os.path.join(DIR, '../data', 'reaches.shp')
-BASIN_PATH = os.path.join(DIR, '../data', 'basins.shp')
-CENTROID_PATH = os.path.join(DIR, '../data', 'centroids.shp')
-CONFUL_PATH = os.path.join(DIR, '../data', 'confluences.shp')
+REACH_PATH = os.path.join(DIR, "../data", "reaches.shp")
+BASIN_PATH = os.path.join(DIR, "../data", "basins.shp")
+CENTROID_PATH = os.path.join(DIR, "../data", "centroids.shp")
+CONFUL_PATH = os.path.join(DIR, "../data", "confluences.shp")
+
+# Load expected fields from JSON file
```

Comment:
The vector files validation should be put into the Pyromb library. The Builder Class should responsible for making the fields and geometry is satisfactory to build the objects. The intent the interface (app.py, QGIS plugin) remains light weight, only being used to feed in shapefiles. Pyromb should do the rest. This reduces the coupling between Pyromb and the calling application. When the Builder is constructed, the EXPECTED_FIELDS could be created and assigned as attribute of the Builder Object.